src/Kernel/RectorKernel.php

```diff
@@ -27,10 +28,16 @@ final class RectorKernel
     public function __construct()
     {
         // while running tests we use different DI containers a lot,
-        // therefore make sure we don't compile them over and over again.
-        if (StaticPHPUnitEnvironment::isPHPUnitRun()) {
-            $this->dumpFileCache = true;
+        // therefore make sure we don't compile them over and over again on rector-src.
+        if (! StaticPHPUnitEnvironment::isPHPUnitRun()) {
+            return;
         }
+
+        if (VersionResolver::PACKAGE_VERSION !== '@package_version@') {
```

Comment:
is this condition the "proper" way to detect a prefixed build?

Reply:
The prefixed build got random value for it, so it is a quick solution imo, feel free to provide a better way if you have

Reply:
here you are https://github.com/rectorphp/rector-src/pull/3869